extensions/src/AWSSDK.Extensions.NETCore.Setup/ClientFactory.cs

```diff
@@ -203,67 +206,191 @@ private static AWSCredentials CreateCredentials(ILogger logger, AWSOptions optio
         /// </summary>
         /// <param name="options"></param>
         /// <returns></returns>
-        private static ClientConfig CreateConfig(Type serviceInterfaceType, AWSOptions options)
+        private ClientConfig CreateConfig(AWSOptions options)
         {
-            var configTypeName = serviceInterfaceType.Namespace + "." + serviceInterfaceType.Name.Substring(1) + "Config";
-            var configType = serviceInterfaceType.GetTypeInfo().Assembly.GetType(configTypeName);
+#if NET8_0_OR_GREATER
+            ClientConfig config = T.CreateDefaultClientConfig();
+#else
+            var configTypeName = typeof(T).Namespace + "." + typeof(T).Name.Substring(1) + "Config";
+            var configType = typeof(T).GetTypeInfo().Assembly.GetType(configTypeName);
 
             var constructor = configType.GetConstructor(EMPTY_TYPES);
             ClientConfig config = constructor.Invoke(EMPTY_PARAMETERS) as ClientConfig;
+#endif
 
-            if(options == null)
+            if (options == null)
             {
                 options = new AWSOptions();
             }
 
+            var defaultConfig = options.DefaultClientConfig;
+
+
+            // There is intertwined logic between ServiceURL, Region and DefaultConfigurationMode
+            // in the SDK. They are handled at the start together to make it easier to debug SDK behavior.
+            if (defaultConfig.ServiceURL != null)
+            {
+                config.ServiceURL = defaultConfig.ServiceURL;
+            }
+            // Setting RegionEndpoint only if ServiceURL was not set, because ServiceURL value will be lost otherwise
+            if (options.Region != null && string.IsNullOrEmpty(defaultConfig.ServiceURL))
+            {
+                config.RegionEndpoint = options.Region;
+            }
             if (options.DefaultConfigurationMode.HasValue)
             {
                 config.DefaultConfigurationMode = options.DefaultConfigurationMode.Value;
             }
 
-            var defaultConfig = options.DefaultClientConfig;
-            var emptyArray = new object[0];
-            var singleArray = new object[1];
 
-            var clientConfigTypeInfo = options.DefaultClientConfig.GetType();
-            var properties = clientConfigTypeInfo.GetProperties(BindingFlags.Public | BindingFlags.Instance);
-            foreach (var property in properties)
+
+            if (defaultConfig.AllowAutoRedirect.HasValue)
             {
-                if (property.GetMethod != null && property.SetMethod != null)
```

Comment:
This was all of the blanket generic reflection code that used to be used to automatically use any properties on client configs. Below where the `property.GetMethod.Invoke(defaultConfig, emptyArray);` was used to call the getter this often caused problems because the getters on the ClientConfig often have lazy evaluated logic that would get messed up here.